lib/run_loop/plist_buddy.rb

```diff
@@ -26,6 +26,16 @@ def plist_read(key, file, opts={})
       end
     end
 
+    # !!Experimental!! find device in xcuserstate file.
+    # @param [String] file the plist to read
+    # @return [String] the UDID of device
+    def plist_find_device(file)
+      `#{plist_buddy} -c Print "#{file}"`
+          .encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '')
+          .split("\n")
+          .select { |v| v =~ /dvtdevice.*:/ }[0]
+    end
+
```

Comment:
I don't think this method belongs here; it should be in the detect_aut/xcode.rb file.

I wish we could add something like this to PlistBuddy, but this class it not well written which means it is difficult to extend.

We need to be very careful about shelling out.  There is a new module in the repo:  RunLoop::Shell.

```
module RunLoop
  module DetectAUT
    module Xcode
       
        require "run_loop/shell"
        include RunLoop::Shell

        def some_method_that_shells_out
            args = ["/usr/libexec/PlistBuddy", "-c", "Print #{plist}"]
            options = {:log_cmd => true}
            
            # exec is provided by Shell
            hash = exec(args, options)
            
             # The :out will be coerced by exec to UTF-8
             file_contents = hash[:out]

              # Always check if a String is nil before calling `split` on it.

             # Use `detect` or `find`
             lines.detect do |line|
                # Use this pattern to match vs. `=~`
                line[/dvtdevice/, 0]
             end    
             ...
        end
```